lib/AST/RequirementMachine/RequirementLowering.h

```diff
@@ -104,7 +104,12 @@ struct RuleBuilder {
 
   /// New rules derived from requirements written by the user, which can be
   /// eliminated by homotopy reduction.
-  std::vector<std::pair<MutableTerm, MutableTerm>> RequirementRules;
+  std::vector<std::tuple<MutableTerm, MutableTerm, Optional<unsigned>>>
```

Comment:
I prefer to define a named type if I need a tuple of length > 2 because `std::get<N>()` is kind of ugly. Maybe RequirementRule or something?